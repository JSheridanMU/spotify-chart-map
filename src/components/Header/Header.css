.color-nav {
    background-color: black;
    height: 58px;
}

.btn {
    background-color: transparent;
    border-radius: 2em !important;
    border: 0.2em solid #1DB954 !important;
    color: #1DB954 !important;
    cursor: pointer;
    font-size: 16px !important;
    padding: 0.7em 1.5em;
    text-transform: uppercase;
    transition: all 0.25s ease;
    height: 44px;
  }
  
  .btn:hover {
    background: #1DB954;
    color: #333 !important;
  }
  
  .btn--login {
    margin: 0 auto;
  }