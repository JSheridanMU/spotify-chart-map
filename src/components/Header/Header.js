import React from "react";
import "./Header.css";
import Navbar from "react-bootstrap/Navbar";
import hash from "./hash";

const authEndpoint = 'https://accounts.spotify.com/authorize';
const clientId = process.env.REACT_APP_CLIENT_ID
const redirectUri = process.env.REACT_APP_REDIRECT_URI;
const scopes = [
  "streaming",
  "user-read-email",
  "user-read-private",
  "user-read-playback-state",
  "user-library-read",
  "user-library-modify",
  "user-read-currently-playing",
  "user-modify-playback-state",
];

export default function Header(props) {
  let _token = hash.access_token;

  if(_token){
    props.tokenChange(_token)
  }
  
  return (
    <Navbar className="color-nav" variant="dark">
      <Navbar.Brand href="#home">
        <img
          alt=""
          src="/logo.png"
          width="30"
          height="30"
          className="d-inline-block align-top"
        />{" "}
        Spotify Chart Map
      </Navbar.Brand>
      {!_token && (
            <a
              className="btn btn--loginApp-link ml-auto"
              href={`${authEndpoint}?client_id=${clientId}&redirect_uri=${redirectUri}&scope=${scopes.join(
                "%20"
              )}&response_type=token&show_dialog=true`}
            >
              Login to Spotify
            </a>
          )}
    </Navbar>
  );
}
