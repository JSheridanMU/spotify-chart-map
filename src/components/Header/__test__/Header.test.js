import * as React from "react";
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Header from '.././Header';

afterEach(cleanup)

test('renders without crashing', () => {
  render(<Header/>);
});

test('renders title and button', () => {
  const {getByText} = render(<Header/>);

  getByText('Spotify Chart Map')
  getByText('Login to Spotify')
});

test('matches snapshot', () => {
    const tree = renderer.create(<Header/>).toJSON()
    expect(tree).toMatchSnapshot()
});