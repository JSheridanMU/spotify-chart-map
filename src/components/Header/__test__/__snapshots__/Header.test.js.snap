// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="color-nav navbar navbar-expand navbar-dark"
>
  <a
    className="navbar-brand"
    href="#home"
  >
    <img
      alt=""
      className="d-inline-block align-top"
      height="30"
      src="/logo.png"
      width="30"
    />
     
    Spotify Chart Map
  </a>
  <a
    className="btn btn--loginApp-link ml-auto"
    href="https://accounts.spotify.com/authorize?client_id=d6aa49c0c90c42a49e4aa1e2b04dfb63&redirect_uri=http://localhost:3000&scope=streaming%20user-read-email%20user-read-private%20user-read-playback-state%20user-library-read%20user-library-modify%20user-read-currently-playing%20user-modify-playback-state&response_type=token&show_dialog=true"
  >
    Login to Spotify
  </a>
</nav>
`;
